gatsby/plonerestapi.rst

```diff
@@ -0,0 +1,142 @@
+
+We will be working a lot with this API while working on the Gatsby source-plugin, so it is recommended that you have an API browser to explore the API.
+Install `Postman <https://www.getpostman.com/>`_ and do go through the quick guide to working with plone.restapi:
+
+https://plonerestapi.readthedocs.io/en/latest/exploring.html#exploring-api-postman-onboarding
+
+.. note::
+
+  We will use the same endpoints for loading the site in a browser, but set the header ``Accept: application/json``.
+  This header tells the endpoint to return JSON data in the response for us to process.
+
+Exploring The Plone REST.API
+----------------------------
+
+We will use https://plonedemo.kitconcept.com/en as our source Plone site, since it's already been configured with the plone.restapi and is all ready for our usage.
+
+Let us start with the root itself.
+Send a GET request to https://plonedemo.kitconcept.com/en.
+This returns the JSON data for the root of the Plone site.
+
+.. code-block:: json
+
+  {
+    "@components": {
+        "breadcrumbs": {
+            "@id": "https://plonedemo.kitconcept.com/en/frontpage/@breadcrumbs"
+        },
+        "navigation": {
+            "@id": "https://plonedemo.kitconcept.com/en/frontpage/@navigation"
+        },
+        "workflow": {
+            "@id": "https://plonedemo.kitconcept.com/en/frontpage/@workflow"
+        }
+    },
+    "@id": "https://plonedemo.kitconcept.com/en/frontpage",
+    "@type": "Document",
+    "UID": "5938b3c2c0e147b18fbf20d32842eb06",
+    "allow_discussion": null,
+    "changeNote": "",
+    "contributors": [],
+    "created": "2018-10-13T13:25:31+00:00",
+    "creators": [
+        "admin"
+    ],
+   ...
+
+Let us explore the ``items`` array from the response and click on https://plonedemo.kitconcept.com/en/frontpage.
+We see that it gives a similar response as we got for the root.
+This way all the content objects have equivalent JSON data which our plugin can process and use to create nodes.
+
+
+Exercise
+++++++++
+
+Create a node for the Plone document at https://plonedemo.kitconcept.com/en/demo/a-page.
+Test the node created from the retrieved data by displaying some data in the ``index`` or any other page.
+
+Hints: Use Postman to check the data from the endpoint and process accordingly. 
+The Axios library can be used for handling HTTP requests.
+
+.. note::
+
+    Minor errors may arise when Gatsby node specific fields are overwritten by spreading the Plone object data.
+    For now, they can be fixed by defaulting them to the node field data: `node.parent` is an empty string and `node.children` an empty array.
+    
+
+..  admonition:: Solution
+    :class: toggle
+
+    .. code-block:: javascript
+
+    exports.sourceNodes = async ({ actions }) => {
+      const { createNode } = actions;
+
+      const { data } = await axios.get('https://plonedemo.kitconcept.com/en/demo/a-page', {
```

Comment:
i see some problems for a newbie.

You mentioned axios in the description: i would give a link to npm or github, so they can directly access to its documentation.

You are using async method here for the first time. You should probably explain what it is and how it works (python developers are not used to async methods)

Could be a good idea to provide a link to official gatsby documentation about sourceNodes?